src/posix/platform/netif.cpp

```diff
@@ -505,6 +517,207 @@ static void UpdateLink(otInstance *aInstance)
     }
 }
 
+#if OPENTHREAD_POSIX_CONFIG_INSTALL_EXTERNAL_ROUTES_ENABLE
+otError AddRtAttr(struct nlmsghdr *aHeader, uint32_t aMaxLen, uint8_t aType, const void *aData, uint8_t aLen)
+{
+    uint8_t        len = RTA_LENGTH(aLen);
+    struct rtattr *rta;
+    otError        error = OT_ERROR_NONE;
+
+    VerifyOrExit(NLMSG_ALIGN(aHeader->nlmsg_len) + RTA_ALIGN(len) <= aMaxLen, error = OT_ERROR_NO_BUFS);
```

Comment:
```suggestion
    OT_ASSERT(NLMSG_ALIGN(aHeader->nlmsg_len) + RTA_ALIGN(len) <= aMaxLen);
```
I think it should never fail.